### PR TITLE
Caching Docs [draft]

### PR DESCRIPTION
**Pull request by yschimke**
All advise, welcome.  Please apply any and all relevant edits directly.

### CODE REVIEW

**Comment by yschimke** [comment]
It's probably useful if we tie the documentation to EventListener, make it tangible to follow.  n.b. I tried to understand cache behaviour in a real world app, and had a lot of questions.  Following up here and with event listener.

**Comment by yschimke** [comment]
Related questions - https://github.com/square/okhttp/issues/6022

File: docs/caching.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+ - CallStart
+ - **CacheHit**
+ - CallEnd
+ 
+### Cache Miss
+
+Under a cache miss, the normal request events are seen, but an additional event shows the presence of the cache.
+
+ - CallStart 
+ - **CacheMiss**
```
Comment (swankjesse) [inline docs/caching.md:43]: Worth a brief overview of potential causes of a cache miss?

File: docs/caching.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+ - ProxySelectStart
+ - ... Standard Events ...
+ - CallEnd
+        
+### Conditional Cache Hit
+ 
+When cache flags, require checking the cache results are still valid, an early cacheConditionalHit event is
```
Comment (swankjesse) [inline docs/caching.md:50]: more commas

Comment (swankjesse) [reply docs/caching.md:50]: also I think this sentence wants to be reworked

File: docs/caching.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+received, followed by a cache hit or miss.  Critically in the cache hit scenario, the response body will be 0 bytes.
```
Comment (swankjesse) [inline docs/caching.md:51]: the response body will be 0 bytes -> the server won’t send the response body

Comment (swankjesse) [reply docs/caching.md:51]: (the response body is non-empty, it’s just not transmitted)

File: docs/caching.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+The response will have non-null cacheResponse and networkResponse. The cacheResponse will be used as the top level
```
Comment (swankjesse) [inline docs/caching.md:53]: put these words in code font? `cacheResponse` and `networkResponse`.

File: docs/caching.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+response only if the response code is HTTP_NOT_MODIFIED (304).
```
Comment (swankjesse) [inline docs/caching.md:54]: I think you wanted this:
HTTP/1.1 304 Not Modified.

(There’s no such symbol HTTP_NOT_MODIFIED)

File: docs/caching.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+Deleting the cache when it is no longer needed can be done.  However this may delete the purpose of the cache
+which is designed to persist between app restarts.
+
+```
+cache.delete()
+```
+ 
+## Pruning the Cache
+
+Pruning the entire Cache to clear space temporarily, can be done using evictAll.
```
Comment (swankjesse) [inline docs/caching.md:78]: commas

File: docs/caching.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+```
+cache.evictAll()
+```
+
+Removing individual items can be done using the urls iterator.
```
Comment (swankjesse) [inline docs/caching.md:84]: Maybe mention that you’ll need to do this when your application knows a cache entry is stale, even though it thinks it is fresh?

**Comment by yschimke** [comment]
Uggghhhhhhhhhhh!!!!!!!!!!

Committed with [draft] in the title.

**Pull request merged**